kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala

```diff
@@ -329,103 +330,172 @@ class DeltaTableWritesSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBa
-
-      val schema = new StructType()
-        .add("id", INTEGER)
-        .add("Part1", INTEGER) // partition column
-        .add("part2", INTEGER) // partition column
-
-      val txn = txnBuilder
-        .withSchema(engine, schema)
-        // partition columns should preserve the same case the one in the schema
-        .withPartitionColumns(engine, Seq("part1", "PART2").asJava)
-        .build(engine)
-
-      assert(txn.getSchema(engine) === schema)
-      // Expect the partition column name is exactly same as the one in the schema
-      assert(txn.getPartitionColumns(engine) === Seq("Part1", "part2").asJava)
-      val txnResult = txn.commit(engine, emptyIterable())
-
-      assert(txnResult.getVersion === 0)
-      assertCheckpointReadiness(txnResult, isReadyForCheckpoint = false)
-
-      verifyCommitInfo(tablePath, version = 0, Seq("Part1", "part2"))
-      verifyWrittenContent(tablePath, schema, Seq.empty)
-    }
-  }
-
-  test("create table with all supported types") {
-    withTempDirAndEngine { (tablePath, engine) =>
-      val parquetAllTypes = goldenTablePath("parquet-all-types")
-      val schema = removeUnsupportedTypes(tableSchema(parquetAllTypes))
-
-      val table = Table.forPath(engine, tablePath)
-      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
-      val txn = txnBuilder.withSchema(engine, schema).build(engine)
-      val txnResult = txn.commit(engine, emptyIterable())
+  Seq(true, false).foreach(
+    executePostCommitHook =>
+      test("create a partitioned table" + {
+        if (executePostCommitHook) " and execute post commit hook" else ""
+      }) {
+        withTempDirAndEngine {
+          (tablePath, engine) =>
+            val table = Table.forPath(engine, tablePath)
+            val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+            val schema = new StructType()
+              .add("id", INTEGER)
+              .add("Part1", INTEGER) // partition column
+              .add("part2", INTEGER) // partition column
+
+            val txn = txnBuilder
+              .withSchema(engine, schema)
+              // partition columns should preserve the same case the one in the schema
+              .withPartitionColumns(engine, Seq("part1", "PART2").asJava)
+              .build(engine)
+
+            assert(txn.getSchema(engine) === schema)
+            // Expect the partition column name is exactly same as the one in the schema
+            assert(txn.getPartitionColumns(engine) === Seq("Part1", "part2").asJava)
+            val txnResult = txn.commit(engine, emptyIterable())
+
+            assert(txnResult.getVersion === 0)
+            assertCheckpointReadiness(txnResult, isReadyForCheckpoint = false)
+            assertChecksumSimpleReadiness(txnResult)
+            if (executePostCommitHook) {
+              txnResult.getPostCommitHooks.forEach(hook => hook.threadSafeInvoke(engine))
```

Comment:
here we are exeucting the hooks, but not checking anything?

Wondering if there is a better way to do this.